src/main/webapp/mapScript.js

```diff
@@ -25,21 +35,86 @@ function initMap() {
 function geocodeAddress(geocoder, resultsMap) {
     const address = document.getElementById('address').value;
     geocoder.geocode({'address': address}, (results, status) => {
-    if (status === 'OK') {
-        resultsMap.setCenter(results[0].geometry.location);
+        if (status === 'OK') {
+            resultsMap.setCenter(results[0].geometry.location);
+        } else {
+            alert('Geocode was not successful for the following reason: ' + status);
+        }
+    });
+}
+
+/**
+ * Determines which group was checked or unchecked
+ * Then sends that group to either be removed from the map if unchecked
+ * Or added to the map if checked
+ */
+function addGroupChecked(checkbox, groupId) {
+    // Get the group name from the checkbox that was checked 
+    if (checkbox.checked) {
+        window.alert('Checked group:' + groupId);
+        addGroupMarkers(groupId);
     } else {
-        alert('Geocode was not successful for the following reason: ' + status);
+        window.alert('Unchecked group:' + groupId);
+        removeGroupMarkers(groupId);
     }
-    });
 }
 
+/**
+ * Adds a group and its events to the map based on the group passed
+ */
+function addGroupMarkers(groupId) {
+    // Get event's with GroupID 
+    // For each event in the group 
+    // Call getMarkerInfo on eventId
+}
+
+/**
+ * Removes the markers of the given group from the map
+ */
+function removeGroupMarkers(groupId) { 
+    // For each event in the group,
+    // hide each of the markers on map
+}
+
+/**
+ * Fetch all the events to add to the map 
+ */
+function getMarkerInfo(eventId, geocoder, map) {
+    // Fetch the json with the eventMarker objects 
+    // For each object call addMarker()
+    const address = 'San Diego, CA'; 
+    const description = 'This is the hardcoded test description';
+    const name = 'Test event';
+    
+    geocoder.geocode({'address': address}, (results) => {
+        // Geocoded the address to longitude and lattitude for
+        // Marker placement
+        addMarker(results[0].geometry.location, description, name, map);
+    });
+}
 
 /**
  * Add markers onto the map 
  * Using the eventID get the name and description for
  * the infoWindow 
  */
-function addMarker(eventID, address, geocoder) {
-    // TODO call event getter methods with eventID 
-    // Will get the name and description to add to the infoWindow
+function addMarker(address, descriptionText, nameText, map) {
+    const description = goog.dom.createDom(goog.dom.TagName.P);
+    goog.dom.setTextContent(description, descriptionText); 
+
+    const name = goog.dom.createDom(goog.dom.TagName.H2);
+    goog.dom.setTextContent(name, nameText) 
+
+    const infoContent =  goog.dom.createDom(goog.dom.TagName.DIV, null, name, description);
```

Comment:
Can you make a template to represent this DOM creation? It's easier to read. Then you can render the template then pass it to InfoWindow.